include/url.h

```diff
@@ -79,6 +79,22 @@ namespace Url
             , fragment_(other.fragment_)
             , userinfo_(other.userinfo_) { }
 
+        /**
+         * Take on the value of the other URL.
+         */
+        Url& assign(const Url& other);
```

Comment:
In fact, `assign` delegates out to the default `operator=` below, but `assign` is present to be more accessible from Cython.